src/routes/working-hours_.$employeeId.work-shifts.$workShiftId.details.tsx

```diff
@@ -46,18 +63,167 @@ function WorkShiftDetails() {
       results.map((result) => result.data).filter(DataValidation.validateValueIsNotUndefinedNorNull),
   });
 
+  const workShiftStartedAt = useMemo(
+    () => DateTime.fromJSDate(workShiftQuery.data?.startedAt ?? new Date(Date.now())),
+    [workShiftQuery.data],
+  );
+
+  /**
+   * Fallback time for shift end is 24 hours after shift start
+   */
+  const workShiftEndedAt = useMemo(
+    () => workShiftQuery.data?.endedAt ?? workShiftStartedAt.plus({ hours: 24 }).toJSDate(),
+    [workShiftQuery.data, workShiftStartedAt],
+  );
+
+  const truckLocationsQuery = useQueries({
+    queries:
+      trucksQuery.map(({ id }) => ({
+        queryKey: ["truck-locations"],
+        queryFn: async () =>
+          id
+            ? {
+                truckId: id,
+                locations: await api.trucks.listTruckLocations({
+                  truckId: id,
+                  after: workShiftQuery.data?.startedAt,
+                  before: workShiftEndedAt,
+                  max: 100000,
+                }),
+              }
+            : null,
+      })) ?? [],
+    combine: (results) => ({
+      data: results
+        .map((result) => {
+          const maxTimestamp =
+            result.data?.locations === undefined
+              ? 0
+              : Math.max(...result.data.locations.map((location) => location.timestamp));
+          const minTimestamp =
+            result.data?.locations === undefined
+              ? 0
+              : Math.min(...result.data.locations.map((location) => location.timestamp));
```

Comment:
Test before applying
```suggestion
          const locationTimestamps = result.data.locations
            ?.map((location) => location.timestamp)
            .toSorted((a, b) => (a ?? 0) - (b ?? 0));
          const minTimestamp = locationTimestamps?.at(0) ?? 0;
          const maxTimestamp = locationTimestamps?.at(-1) ?? 0;
```